sdk/include/thread.h

```diff
@@ -23,6 +23,17 @@ typedef struct
 [[cheri::interrupt_state(disabled)]] SystickReturn __cheri_compartment("sched")
   thread_systemtick_get(void);
 
+enum ThreadSleepFlags : uint32_t
+{
+	/**
+	 * Sleep for up to the specified timeout, but wake early if there are no
+	 * other runnable threads.  This allows a high-priority thread to yield for
```

Comment:
```suggestion
	 * other threads runnable or blocked with earlier timeouts.
	 * This allows a high-priority thread to yield for
```

I'm trying to capture the difference between "wake as soon as the system idles within the next N ticks" (which would be the "traverse the waiting queue until we found a yielding thread" behavior) and "wake as soon as the system looks like it might be idle N ticks from now" (the "only test the head of the waiting queue for yielding" behavior now implemented).

Reply:
I've added some more explanation.